services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs

```diff
@@ -455,6 +482,49 @@ impl SentinelS2L2aCogsMetaData {
         }
         debug!("number of generated loading infos: {}", parts.len());
 
+        // This is a workaround to avoid errors when a provider does not fill the complete query rectangle.
+        let buffered_start =
+            query.time_interval.start() + Duration::seconds(self.stac_query_buffer.start_seconds);
+        let buffered_end =
+            query.time_interval.end() + Duration::seconds(self.stac_query_buffer.end_seconds);
+
+        if let Some(first) = parts.first() {
```

Comment:
is this the same logic as in gdal and mock source? Maybe we should reuse the code

Reply:
logic still is similar however, i don't know how to reuse in a good way